File: various/oom-lab1/FormBuilder.java
package lab1;

public interface FormBuilder {
	public Object startBuilding();
	public void endBuilding();
	public Object startGroup(Object parent, String label);
	public void endGroup(Object parent);
	public void buildCheckbox(Object parent, String label);
	public Object startRadioGroup(Object parent);
	public void buildRadioButton(Object radioGroup, String label);
	public void endRadioGroup(Object parent);
	public void buildText(Object parent, String label);
}

File: c#/builder/builder/FormBuilder.cs
using System;

namespace builder
{
	public interface FormBuilder
	{
		Object StartBuilding();
		void EndBuilding();
		Object StartGroup(Object parent, string label);
		void EndGroup(Object parent);
		void BuildCheckbox(Object parent, string label);
		Object StartRadioGroup(Object parent);
		void BuildRadioButton(Object radioGroup, string label);
		void EndRadioGroup(Object parent);
		void BuildText(Object parent, string label);
	}
}
